internal/util/cephcmds.go

```diff
@@ -32,6 +32,42 @@ import (
 // InvalidPoolID used to denote an invalid pool.
 const InvalidPoolID int64 = -1
 
+// ExecuteCommandWithNSEnter executes passed in program with args with nsenter
+// and returns separate stdout and stderr streams. In case ctx is not set to
+// context.TODO(), the command will be logged after it was executed.
+func ExecuteCommandWithNSEnter(ctx context.Context, netPath, program string, args ...string) (string, string, error) {
+	var (
+		sanitizedArgs = StripSecretInArgs(args)
+		stdoutBuf     bytes.Buffer
+		stderrBuf     bytes.Buffer
+	)
+	//  nsenter --net=%s -- <program> <args>
+	args = append([]string{fmt.Sprintf("--net=%s", netPath), "--", program}, args...)
```

Comment:
In order to improve security, please check that the `netPath` exists. The path is read from a ConfigMap, which could contain some unexpected strings. Not sure what happens if `netPath` is set to `/proc/self/ns/net -- /bin/date ; echo DO NOT DO THIS`

Reply:
added a check for the path exists, if self is given it will be executed in its own namespace?

Reply:
yes, `self` is a symlink to the process' own PID.